### PR TITLE
Cockpit 250

### PR DESCRIPTION
**Pull request by garrett**
Preview @ https://garrett.github.io/cockpit-project.github.io/blog/cockpit-250.html

TODO:

- [ ] Make sure alert screenshot is not in a responsive wrapping mode (it should be wide enough that the button is to the right, not underneith)
- [ ] Resize the mobile screenshot so it's 1 pixel to 1 pixel, not HiDPI version
- [ ] Re-take migration screenshot (it's out of date, as we no longer have the shared vs. copy option)
- [ ] Text editing

### CODE REVIEW

**Comment by garrett** [comment]
Perhaps we should mention https://github.com/cockpit-project/cockpit/pull/16169 and https://github.com/cockpit-project/cockpit/issues/16173 (when it is implemented and lands, hopefully today) in the release notes?

**Comment by martinpitt** [comment]
@garrett : Agreed -- I added release-note label to PR #16169 (of course the actual note still needs to be written).

**Comment by garrett** [comment]
Oh, both of these changes landing in Cockpit are the result of @Verhoeckx bringing up the points in the Cockpit discussion area. Additionally, we have made it so one cannot lock themselves out of Cockpit too, also thanks to one of his discussion posts. I think we should add a shout-out (& thanks!) in the release notes too.

And perhaps we should also mention the accounts change? https://github.com/cockpit-project/cockpit/issues/16152

**Comment by marusak** [comment]
> And perhaps we should also mention the accounts change? cockpit-project/cockpit#16152

I don't think we should - it is a bit of weird one - "Accounts: You cannot lock yourself out anymore". I don't think that anyone was using this so we should announce it as we might break someone workflow, nor that it was actively causing issues for someone.

**Comment by garrett** [comment]
> we might break someone workflow

[workflow](https://xkcd.com/1172/)

**Comment by Verhoeckx** [comment]
> Oh, both of these changes landing in Cockpit are the result of @Verhoeckx bringing up the points in the Cockpit discussion area. Additionally, we have made it so one cannot lock themselves out of Cockpit too, also thanks to one of his discussion posts. I think we should add a shout-out (& thanks!) in the release notes too.
> 
> And perhaps we should also mention the accounts change? [cockpit-project/cockpit#16152](https://github.com/cockpit-project/cockpit/issues/16152)

Thanks, that would be really great :smiley: !

**Comment by Verhoeckx** [comment]
Oh, and I think the title should be Cockpit 249?

**Comment by marusak** [comment]
> Oh, and I think the title should be Cockpit 249?

No, this is 250. 249 was released two weeks ago.

**Comment by Verhoeckx** [comment]
Oh sorry, it's not mentioned on the website so I thought it had to be 249.

**Comment by martinpitt** [comment]
@garrett: I updated https://github.com/cockpit-project/cockpit/pull/16169 (screenshots, including the session menu change and credit). I dropped the obsolete release-note label from https://github.com/cockpit-project/cockpit-machines/pull/239.

I also think it's finally time to actually advertise the [machines migration feature](https://github.com/cockpit-project/cockpit-machines/pull/15), as we landed a few fixes now. @skobyda , what do you think? I updated the text of the release note there.

@garrett , do you mind a round of proof-reading and updating the PR?

**Comment by garrett** [comment]
Updated preview

**Comment by garrett** [comment]
Added items to the TODO, for the images mainly

**Comment by garrett** [comment]
FWIW, to run the script, run:

```sh
_scripts/toolbox-ruby _scripts/generate-release-notes.rb
```

This assumes you have a toolbox container for the website. If not, then run this first:

```sh
_scripts/toolbox-create
```

File: _posts/2021-08-03-cockpit-250.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+All of the above changes only affect "limited access" mode. Normal administration mode looks the same as before.
+
+The menu on the top right is now called "Session" instead of showing an user icon. This clarifies that the settings in that menu apply to remote hosts as well, which possibly run as a different user.
+
+*Thanks to Verhoeckx for suggesting these design changes!*
+
+## Machines: Migration of a virtual machine
+
+The Machines page can now migrate a VM to a different machine. It supports live, offline, and temporary migration. Commonly, the VM's storage is on shared storage, which avoids having to copy the data to the remote machine and makes the migration process much faster and more robust. As it is not possible to automatically detect whether the storage is shared, the user has to explicitly tell whether the storage must be copied.
```
Comment (skobyda) [inline _posts/2021-08-03-cockpit-250.md:48]:
```suggestion
The Machines page can now migrate a VM to a different machine. It supports live, and temporary migration. Commonly, the VM's storage is on shared storage, which avoids having to copy the data to the remote machine and makes the migration process much faster and more robust. As it is not possible to automatically detect whether the storage is shared, the user has to explicitly tell whether the storage must be copied.
```
The offline migration is not supported.

**Review by skobyda — CHANGES_REQUESTED** [review]

File: _posts/2021-08-03-cockpit-250.md (hunk 1 above)
Comment (martinpitt) [reply _posts/2021-08-03-cockpit-250.md:48]: Thanks @skobyda -- I updated the text in https://github.com/cockpit-project/cockpit-machines/pull/15 as that is the primary truth, and I'll update this PR once again (the second PR with the "Session" menu slipped in)

**Comment by Verhoeckx** [comment]
> Thanks to Verhoeckx for suggesting these design changes!

To be honest it was actually you who came up with the real designs. I only reported the problems that I had. Having said that, I like my name mentioned in the release notes :smiley: !

**Comment by martinpitt** [comment]
I updated the text in all PRs, optipng/pngquanted the screenshots, fixed the labels, and re-ran the generation script. However, for some reason the preview doesn't update -- as Garrett is out, I better file my own PR, and we continue the review/discussion there.

**Pull request closed without merging**